src/setup.rs

```diff
@@ -10,37 +9,21 @@ async fn setup_template_project(year: i32) -> Result<(), AocError> {
         return Err(AocError::SetupExists);
     }
 
-    let res = tokio::process::Command::new("cargo")
-        .args(["new", &format!("year_{}", year)])
-        .output()
-        .await?;
-
-    if !res.status.success() {
-        return Err(AocError::SetupExists);
-    }
-
-    tokio::fs::rename(format!("year_{year}"), format!("{year}")).await?;
+    let year = format!("{}", year);
+    tokio::fs::create_dir(&year).await?;
 
     let template_dir = format!("{}/template", env!("CARGO_MANIFEST_DIR"));
-    let bins = tokio::fs::read(Path::new(&template_dir).join("Cargo.toml.template")).await?;
 
-    OpenOptions::new()
-        .append(true)
-        .open(format!("{}/Cargo.toml", year))
-        .await?
-        .write_all(&bins)
-        .await?;
+    for day in 1..=25 {
+        let day = format!("day_{:0>2}", day);
+        tokio::process::Command::new("cargo")
+            .args(["new", &day])
+            .current_dir(&year)
+            .output()
+            .await?;
 
-    for i in 1..=25 {
-        let dir = format!("{year}/src/day_{:0>2}", i);
-        tokio::fs::create_dir_all(&dir).await?;
-        tokio::fs::copy(
-            Path::new(&template_dir).join("template.rs"),
-            format!("{dir}/main.rs"),
-        )
-        .await?;
+        tokio::fs::copy(format!("{template_dir}/template.rs"), format!("{year}/{day}/main.rs")).await?;
```

Comment:
Should probably be `tokio::fs::copy(format!("{template_dir}/template.rs"), format!("{year}/{day}/src/main.rs")).await?;`

Notice the `src` before `main.rs`. Because, now it looks like this:

```
<img width="528" alt="image" src="https://github.com/user-attachments/assets/ef451e02-c12d-44dd-b5f8-8a9358ed9d86">
```

Where the `main.rs` contains this:
```rs
fn main() {
    println!("Hello, world!");
}
```

Also; what do you feel about formatting the template file as well according to how we have formatted the rest of this codebase?

Reply:
We can do that